test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -46,6 +46,12 @@ class test_ucp_worker_discard : public ucp_test {
         m_ep_test_info_map.clear();
     }
 
+    void cleanup()
+    {
+        ucp_test::cleanup();
+        EXPECT_EQ(m_created_ep_count, m_destroyed_ep_count);
```

Comment:
why needed?
will valgrind show a leak?

Reply:
it is needed to check that `m_created_ep_count == m_destroyed_ep_count` at the end of the testing to make sure that UCP worker destroy works fine

Reply:
can we do it in test_worker_discard?

Reply:
done